SonarQube.TeamBuild.PreProcessor/SonarQubeServerFactory.cs

```diff
@@ -12,14 +12,18 @@ namespace SonarQube.TeamBuild.PreProcessor
 {
     internal class SonarQubeServerFactory : ISonarQubeServerFactory
     {
-        public ISonarQubeServer Create(ProcessedArgs args)
+        public ISonarQubeServer Create(ProcessedArgs args, ILogger logger)
         {
             if (args == null)
             {
                 throw new ArgumentNullException("args");
             }
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
 
-            return new SonarWebService(GetDownloader(args), args.GetSetting(SonarProperties.HostUrl));
+            return new SonarWebService(GetDownloader(args), args.GetSetting(SonarProperties.HostUrl), logger);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) In method 'SonarQubeServerFactory.Create(ProcessedArgs, ILogger)', call System.IDisposable.Dispose on object 'new SonarWebService(SonarQubeServerFactory.GetDownloader(args), args.GetSetting("sonar.host.url"), logger)' before all references to it are out of scope. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3ADisposeObjectsBeforeLosingScope)